rockcraft/parts.py

```diff
@@ -67,6 +69,10 @@ def __init__(
 
         emit.progress("Initializing parts lifecycle")
 
+        if not PartsLifecycle._OVERLAY_CALLBACK_REGISTERED:
+            callbacks.register_configure_overlay(_install_overlay_repositories)
```

Comment:
In the base package exclusion I placed the callback registration in [lifecycle.run](https://github.com/canonical/rockcraft/pull/258/files#diff-2e78c02a1430e75e37cceef3157f4c33ebc785b8693642ebeda5b5d118ec5935R42), before the loop that runs lifecycle more than once. The callback function declaration could stay where it is, only the registration would be done earlier.

Reply:
Updated this, please take a look

Reply:
Yeah, that works too.